### PR TITLE
Snapshot tests shouldn't crash the Spice benchmark test

### PR DESCRIPTION
**Pull request by Sevenannn**
## 🗣 Description

This PR addresses the following issues for the snapshot test added to the benchmark test
* Snapshot test failure won't cause the benchmark process to panic
* Write snapshot test failure results back to spice benchmark result dataset
* Propagate both query failure and snapshot failure in benchmark test logs

After this change, snapshot test failure will look like the following:
image

### CODE REVIEW

**Comment by Sevenannn** [comment]
- Close https://github.com/spiceai/spiceai/issues/2605

File: crates/runtime/benches/bench.rs [hunk 1]
```diff
@@ -293,17 +312,30 @@ async fn record_explain_plan(
...
+            insta::assert_snapshot!(format!("{connector}_{query_name}_explain"), explain_plan);
+        });
+        if result.is_err() {
+            assertion_err = Some(format!("Snapshort assertion failed for {connector}, {query_name}"));
+        }
     });
+
+    if let Some(assertion_err) = assertion_err {
+        return Err(assertion_err);
+    }
```
Comment (phillipleblanc) [inline crates/runtime/benches/bench.rs:394]:
```suggestion
    assertion_err.ok_or_else(Err)?;
```

Does this work? (I haven't tested it)

File: crates/runtime/benches/bench.rs [hunk 2]
```diff
@@ -243,8 +247,23 @@ async fn run_query_and_record_result(
...
+    if query_err.is_some() {
+        return Err(format!("Query Error: {}", query_err.unwrap_or_default()));
+    }
+
+    if snapshot_err.is_some() {
+        return Err(format!(
+            "Snapshot Test Error: {}",
+            snapshot_err.unwrap_or_default()
+        ));
     }
```
Comment (phillipleblanc) [inline crates/runtime/benches/bench.rs]:
```suggestion
    match (query_err, snapshot_err) {
    (Some(query), Some(snapshot)) => {
        return Err(format!("Query Error: {}; Snapshot Test Error: {}", query, snapshot));
    }
    (Some(query), None) => {
        return Err(format!("Query Error: {}", query));
    }
    (None, Some(snapshot)) => {
        return Err(format!("Snapshot Test Error: {}", snapshot));
    }
    (None, None) => {}
  }
```

File: crates/runtime/benches/bench.rs (hunk 1 above)
Comment (Sevenannn) [reply crates/runtime/benches/bench.rs:394]: `ok_or_else` only works as follows
```
    assertion_err
        .clone()
        .ok_or_else(|| assertion_err.unwrap_or_default())?;
```

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**